src/Providers/Http/DTO/RequestOptions.php

```diff
@@ -0,0 +1,351 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function fromArray(array $array): self
+    {
+        $timeout = $array[self::KEY_TIMEOUT] ?? null;
+        $connectTimeout = $array[self::KEY_CONNECT_TIMEOUT] ?? null;
+        $allowRedirects = $array[self::KEY_ALLOW_REDIRECTS] ?? null;
+        $maxRedirects = $array[self::KEY_MAX_REDIRECTS] ?? null;
+
+        return new self(
+            $timeout !== null ? (float) $timeout : null,
+            $connectTimeout !== null ? (float) $connectTimeout : null,
+            $allowRedirects !== null ? (bool) $allowRedirects : null,
+            $maxRedirects !== null ? (int) $maxRedirects : null
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function getJsonSchema(): array
+    {
+        return [
+            'type' => 'object',
+            'properties' => [
+                self::KEY_TIMEOUT => [
+                    'type' => ['number', 'null'],
+                    'minimum' => 0,
+                    'description' => 'Maximum duration in seconds to wait for the full response.',
+                ],
+                self::KEY_CONNECT_TIMEOUT => [
+                    'type' => ['number', 'null'],
+                    'minimum' => 0,
+                    'description' => 'Maximum duration in seconds to wait for the initial connection.',
+                ],
+                self::KEY_ALLOW_REDIRECTS => [
+                    'type' => ['boolean', 'null'],
+                    'description' => 'Whether HTTP redirects should be automatically followed.',
+                ],
+                self::KEY_MAX_REDIRECTS => [
+                    'type' => ['integer', 'null'],
+                    'minimum' => 0,
+                    'description' => 'Maximum number of redirects to follow when enabled.',
+                ],
+            ],
+            'additionalProperties' => false,
+        ];
+    }
+
+    /**
+     * Validates timeout values.
+     *
+     * @since n.e.x.t
+     *
+     * @param float|null $value Timeout to validate.
+     * @param string $fieldName Field name for the error message.
+     *
+     * @throws InvalidArgumentException When timeout is negative.
+     */
+    private function validateTimeout(?float $value, string $fieldName): void
+    {
+        if ($value !== null && $value < 0) {
+            throw new InvalidArgumentException(
+                sprintf('Request option "%s" must be greater than or equal to 0.', $fieldName)
+            );
+        }
+    }
+
+    /**
+     * Validates redirect configuration.
+     *
+     * @since n.e.x.t
+     *
+     * @param bool|null $allowRedirects Whether redirects are enabled.
+     * @param int|null $maxRedirects Maximum redirects.
+     *
+     * @throws InvalidArgumentException When redirect count is invalid.
+     */
+    private function validateRedirectLimit(?bool $allowRedirects, ?int $maxRedirects): void
```

Comment:
Do we really want to have `allowRedirects` _and_ `maxRedirects`? I never understood why WordPress has both. IMO setting `maxRedirects` to `0` is a perfectly understandable way of saying "do not allow redirects".

Reply:
Yeah, I agree. Resolved in 6551352